core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/BPMBillingProjectDeleteRunner.scala

```diff
@@ -0,0 +1,131 @@
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Delete a BPM billing project after the job corresponding with jobControlId has finished in WSM
+  * (which will be a landing zone deletion job)
+  * Handles both AzureBillingProjectDelete and OtherBpmBillingProjectDelete
+  *   If it's AzureBillingProjectDelete, it means we need to wait for the associated lz job to complete
+  *   If it's OtherBpmBillingProjectDelete, we can delete right away
+  */
+class BPMBillingProjectDeleteRunner(
+  val samDAO: SamDAO,
+  val gcsDAO: GoogleServicesDAO,
+  workspaceManagerDAO: WorkspaceManagerDAO,
+  billingRepository: BillingRepository,
+  billingLifecycle: BillingProjectLifecycle
+) extends WorkspaceManagerResourceJobRunner
+    with LazyLogging
+    with UserCtxCreator {
+
+  override def apply(
+    job: WorkspaceManagerResourceMonitorRecord
+  )(implicit executionContext: ExecutionContext): Future[JobStatus] = {
+    val projectName = job.billingProjectId match {
+      case Some(name) => RawlsBillingProjectName(name)
+      case None =>
+        logger.error(
+          s"Job to monitor AzureLandingZoneDeleteResult created with id ${job.jobControlId} but no billing project set"
+        )
+        return Future.successful(Complete) // nothing more this runner can do with it
+    }
+    val userEmail = job.userEmail match {
+      case Some(email) => email
+      case None =>
+        logger.error(
+          s"Job to monitor ${job.jobType} $projectName created with id ${job.jobControlId} but no user email set"
+        )
+        val errorMsg =
+          s"Unable to update ${projectName.value} with landing zone deletion status because no user email is stored on monitoring job"
+        return billingRepository
+          .updateCreationStatus(projectName, DeletionFailed, Some(errorMsg))
+          .map(_ => Complete)
+    }
+    getUserCtx(userEmail).transformWith {
+      case Success(userCtx) =>
+        job.jobType match {
+          case AzureBillingProjectDelete =>
+            billingRepository.getLandingZoneId(projectName).flatMap {
+              case Some(landingZoneId) =>
+                handleLandingZoneDeletion(job.jobControlId, projectName, UUID.fromString(landingZoneId), userCtx)
+              case None =>
+                val msg = s"No landing zone id set on Azure billing project"
+                logger.error(msg)
+                billingRepository.updateCreationStatus(projectName, DeletionFailed, Some(msg)).map(_ => Complete)
+            }
+          case _ => billingLifecycle.finalizeDelete(projectName, userCtx).map(_ => Complete)
+        }
+      case Failure(t) =>
+        val msg = s"Unable to complete billing project deletion: unable to retrieve request context for $userEmail"
+        logger.error(s"${job.jobType} job ${job.jobControlId} for billing project: $projectName failed: $msg", t)
+        billingRepository.updateCreationStatus(projectName, DeletionFailed, Some(msg)).map(_ => Incomplete)
```

Comment:
What do you think of keeping the status as `Deleting` but adding the error message?

Alternatively, we could split `DeletionFailed` to two separate statuses, `RetryableDeletionFailure` and `FatalDeletionFailure`.

I'm mostly concerned about our ability to debug billing project deletion failures. It would be nice to be able to tell if Rawls is actively retrying for a given billing project or whether it's given up.

On a separate note, it also might be worth adding a retry limit to keep billing projects from getting endlessly retried